### ISSUE TITLE
Feature request: fortran support (blocked b/c there is no LLVM-based fortran compiler)

### ISSUE THREAD

**Issue by hoodmane** [emscripten-core/emscripten#16601]
A lot of foundational software is written in Fortran, for instance BLAS and LAPACK and parts of scipy. We are currently using f2c + horrendous, cursed hacks to compile scipy:
https://github.com/scipy/scipy/issues/15290
https://github.com/pyodide/pyodide/pull/2289

It would be fantastic to be freed of these cursed hacks. For this, we need a fortran compiler as part of the emscripten toolchain. As far as I can tell, there are three candidates for LLVM-based C compilers: flang classic, LFortran, and flang (which is part of LLVM). As far as I can tell, none of these actually work. But hopefully some day flang will work. When it does, it would be fantastic if we could have an `emflang`.

**Comment by arsnyder16** [comment]
@hoodmane I am currently using BLAS/LAPACK within my wasm library. I took a similar approach described here https://chrz.de/2020/04/21/fortran-in-the-browser/  this has a lot of good details on what the hurdles are for FORTRAN on wasm.

https://github.com/StarGate01/Full-Stack-Fortran 

Side note: If you are doing a lot of scientific libraries the only thing i have yet to be be successful with is omp. I have it compiling under wasm but linking is failing for my library based on what i believe is some missing assembly code that clang is compiling for omp support. I have yet to investigate it much further than that but that is my initial impression of the issue. I have very little experience with omp, so i could be off

**Comment by arsnyder16** [comment]
That post and the code is using emscripten 2, but i was able to successfully get emscripten  3.1.x to work

**Comment by hoodmane** [comment]
Thanks so much @arsnyder16! This looks super interesting. I'll have to see if we can make it work for us. Currently we have to just delete certain functions that use newer Fortran features since f2c can't handle them.

> That post and the code is using emscripten 2, but i was able to successfully get emscripten 3.1.x to work

In Pyodide we just made it up to Emscripten 2.0.27, currently blocked on a Firefox bug for later versions #16538, so Emscripten 2 is fine. My feeling is that Emscripten 3.x isn't hugely different (modulo the Firefox bugs, the changes that take us from 2.0.27 to tip of tree are much smaller than the changes we needed to get from 2.0.16 to 2.0.27 which took a fair amount of work for us).

**Comment by hoodmane** [comment]
Oh but I see you were using Emscripten v1.39.12, which sounds more concerningly old to me.

**Comment by hoodmane** [comment]
@arsnyder16 I see in your "Future Work" section of your blog:

> Integrate this stack into existing projects that need it

Any help you can provide for Pyodide in this regard would be extremely appreciated! I made an issue about it here:
https://github.com/pyodide/pyodide/issues/2318

**Comment by arsnyder16** [comment]
@hoodmane Just to be clear that is not my blog post or github repo, but something i found along the way, that is related to what you are trying to accomplish. 

As far as v1.39.12 that is just the blog post, the github repo show 2.0.12
https://github.com/StarGate01/Full-Stack-Fortran/blob/ab99e3d20391299d6bf651fc31796c812c76b5e2/tools/f90wasm/Dockerfile#L24

I won't be able to provide you with an full stack integration with this, but i can say it was pretty straight forward using this example to produce static libraries which could then be linked successfully into another project

**Comment by hoodmane** [comment]
> that is not my blog post or github repo

Ah right, thanks.

**Comment by hoodmane** [comment]
Closed as duplicate of #5553.
